app/controllers/subscriptions_controller.rb

```diff
@@ -2,20 +2,21 @@ class SubscriptionsController < ApplicationController
   def create
     return render json: { id: 0 }, status: :created if smoke_test_address?
 
-    subscription = send_email = status = nil
+    subscriber = Subscriber.resilient_find_or_create(
+      address,
+      signon_user_uid: current_user.uid,
+    )
 
-    Subscription.transaction do
-      if (subscriber_was_deactivated = subscriber.deactivated?)
-        subscriber.activate!
-      end
+    subscription, email, status = subscriber.with_lock do
+      deactivated_subscriber = subscriber.deactivated?
+      subscriber.activate! if deactivated_subscriber
 
-      existing_subscription = Subscription.active.lock.find_by(
+      existing_subscription = Subscription.active.find_by(
         subscriber: subscriber,
         subscriber_list: subscriber_list,
       )
 
-      create_new_subscription = existing_subscription.nil? ||
-        (frequency.to_s != existing_subscription.frequency.to_s)
+      create_new_subscription = frequency.to_s != existing_subscription&.frequency.to_s
```

Comment:
Do we need to be calling `to_s` on these when they will already be strings (or nil)?

Reply:
I'll check if it's necessary or not.

Reply:
I managed to remove them 🎉